src/compact_live.rs

```diff
@@ -0,0 +1,454 @@
+                &noisy_encrypted_entry_table,
+                &chains.chain_values,
+            )?;
+
+            // Insert all recompacted chains first.
+            self.insert_chain_table(&new_chains.iter().flat_map(|(_, v)| v.clone()).collect()).await?;
+
+            // Try upserting the new entry table.
+            let upsert_data = UpsertData::new(
+                &noisy_encrypted_entry_table,
+                noisy_entry_table.encrypt::<DEM_KEY_LENGTH, DemScheme>(k_value, rng)?,
+            );
+
+            // These are failures to upsert.
+            noisy_encrypted_entry_table = self.upsert_entry_table(&upsert_data).await?;
+
+            // Delete unused chains (at least one chain element per upsert try):
+            let mut chains_to_delete: HashSet<Uid<UID_LENGTH>> =
+                HashSet::with_capacity(upsert_data.len());
+            chains_to_delete.extend(
+                // - new chains corresponding to unsuccessful upserts
+                new_chains
+                    .iter()
+                    .filter_map(|(uid, chains)| {
+                        if noisy_encrypted_entry_table.contains_key(uid) {
+                            Some(chains.keys())
+                        } else {
+                            None
+                        }
+                    })
+                    .flatten()
+                    .cloned(),
+            );
+
+            chains_to_delete.extend(
+                // - old chains corresponding to successful upserts
+                chains.chain_uids
+                    .iter()
+                    .filter_map(|(uid, chain)| {
+                        if !noisy_encrypted_entry_table.contains_key(uid) && !noise.contains(uid) {
+                            Some(chain)
+                        } else {
+                            None
+                        }
+                    })
+                    .flatten()
+                    .cloned(),
+            );
+
+            self.delete_chain(&chains_to_delete).await?;
+        }
+
+        Ok(())
+    }
+
+    /// Compacts a random subset of the Entry Table.
+    ///
+    /// The size of this subset is such that `num_reindexing_before_full_set` compact operations
+    /// are needed to compact all the Entry Table.
+    ///
+    /// - `master_key`                      : Findex master key
+    /// - `num_reindexing_before_full_set`  : number of compact operations needed to compact all
+    /// the Entry Table
+    async fn live_compact(
+        &mut self,
+        master_key: &KeyingMaterial<MASTER_KEY_LENGTH>,
+        num_reindexing_before_full_set: u32,
+    ) -> Result<(), Error<CustomError>> {
+        let mut rng = CsRng::from_entropy();
+
+        let k_value =
+            master_key.derive_dem_key::<DEM_KEY_LENGTH, DemScheme::Key>(ENTRY_TABLE_KEY_DERIVATION_INFO);
+
+        let (noisy_entry_table_uids, noise) = self.select_noisy_uids(&mut rng, num_reindexing_before_full_set).await?;
```

Comment:
I think the `noisy_entry_table_uid` is not necessary, could be `entries_uids` because `noisy` is really similar than `noise` but means something totally different in this case (I prefer to not have it for a better comprehension)

Reply:
I thought it was clearer to add the `noisy_` prefix. This indicates that a variable which names is prefixed by it contains both targeted and noise data.

Reply:
It was not clear for me. I think it's better to have two separate variable (and it avoid duplicate memory too)

Reply:
I changed to `mixt_uids` and `noise_uids` and described in several comments the meaning of `mixt_`.